packages/client/src/workflow-client.ts

```diff
@@ -594,6 +610,31 @@ export class WorkflowClient {
     }
   }
 
+  /**
+   * Lazily downloads the histories of the workflows indicated by the passed in iterator. These
+   * histories may be used in conjunction with replay functionality to verify changes to workflow
+   * code.
+   */
+  public downloadLazily(
```

Comment:
If we change `fetchHistory` to return the workflow ID and run ID, I think we can remove this helper from the top-level client. Just accept either an iterable of workflows or an iterable of histories in the replayer and if you get the former, map there.

I understand there may be options later for async fetch, but since there's not now, let's remove this method.

If we don't remove this method, we should at least call it `lazyFetchHistories` or something that makes it clear it's about histories.